packages/runtime-common/merge-relationships.ts

```diff
@@ -22,8 +22,10 @@ function _formatForMerge(
   let data: Record<string, Relationship | Relationship[]> = {};
 
   for (let [key, value] of Object.entries(resource ?? {})) {
-    if (key.split('.').length === 2 && key.split('.')[1].match(/^\d+$/)) {
-      let name = key.split('.')[0];
+    let keys = key.split('.');
+    if (keys.length > 1 && keys[keys.length - 1].match(/^\d+$/)) {
+      keys.pop();
+      let name = keys.join('.');
```

Comment:
I have to update this because with the previous implementation, we couldn't handle linksToMany fields within compound FieldDef like `trips.countriesVisited.1`.